AdyenDropIn/Modules/PaymentMethodList/PaymentMethodListAssembler.swift

```diff
@@ -18,30 +18,41 @@ internal struct PaymentMethodListAssembler: PaymentMethodListAssemblerProtocol {
     private let componentManager: ComponentManager
     private let context: AdyenContext
     private let configuration: DropInComponent.Configuration
+    private let cardComponentDelegate: CardComponentDelegate?
+    private let partialPaymentDelegate: PartialPaymentDelegate?
 
     // MARK: - Initializers
 
     internal init(
         componentManager: ComponentManager,
         context: AdyenContext,
-        configuration: DropInComponent.Configuration
+        configuration: DropInComponent.Configuration,
+        cardComponentDelegate: CardComponentDelegate?,
+        partialPaymentDelegate: PartialPaymentDelegate?
     ) {
         self.componentManager = componentManager
         self.context = context
         self.configuration = configuration
+        self.cardComponentDelegate = cardComponentDelegate
+        self.partialPaymentDelegate = partialPaymentDelegate
     }
 
     // MARK: - PaymentMethodListAssemblerProtocol
 
     internal func resolvePaymentMethodListRouter() -> PaymentMethodListRouterProtocol {
-        let componentContainerAssembler = ComponentContainerAssembler()
+        let componentContainerAssembler = ComponentContainerAssembler(
+            context: context,
+            configuration: configuration,
+            cardComponentDelegate: cardComponentDelegate,
+            partialPaymentDelegate: partialPaymentDelegate
+        )
         let router = PaymentMethodListRouter(componentContainerAssembler: componentContainerAssembler)
         let viewModel = PaymentMethodListViewModel(
-            router: router,
             context: context,
             componentManager: componentManager,
             configuration: configuration
         )
+        viewModel.delegate = router
```

Comment:
We seem to be always setting delegate indeed. Moving it to init would prevent the caller from forgetting this line.

Reply:
Nice one, change it that way.